routes/libraries.js

```diff
@@ -13,75 +32,114 @@ router
-        fullnessRating,
-        genres
-      );
-      res.json(newLibrary); // TODO: will probably be to the library's page
-    } catch (e) {
-      res
-        .status(500)
-        .render({ errorCode: 500, title: "error", id: "NEED TO FIX" });
+  ;
+router.post("/", (req, res, next) => {
+  try {
+    console.log(req.body)
+    // validation.checkImageFileString(req.body.image, "Libarys Image");
+    next();
+  } catch (e) {
+    // TODO: make it rerender!!!
+    return res.status(400).send(`${e} Error: Invalid file type`);
+  }
+}, upload.single('image'), async (req, res) => { // Currently creates libary and sends json of created library
+  if(!req.file){ // Something went wrong saving the image
+    // TODO: make it rerender!!!
+    return res.status(500).send({ status: "Error", message: "Uh, Oh! Something wrong went on our side, we will fix it soon!" });
+  }
+  console.log(req.body)
+  const newLibraryData = req.body;
+  let errors = [];
+  try {
+    newLibraryData.name = validation.checkString(
+      newLibraryData.name,
+      "Name"
+    );
+  } catch (e) {
+    errors.push(e);
+  }
+  try {
+    newLibraryData.lat = parseInt(newLibraryData.lat);
+    newLibraryData.lat = validation.isValidNumber(
+      newLibraryData.lat,
+      "Librarys Latitude"
+    );
+  } catch (error) {
+    errors.push(e);
+  }
+  try {
+    newLibraryData.lng = parseInt(newLibraryData.lng);
+    newLibraryData.lng = validation.isValidNumber(
+      newLibraryData.lng,
+      "Librarys Longitude"
+    );
+  } catch (error) {
+    errors.push(e);
+  }
+  try {
+    req.user._id= validation.checkValidId(
+      req.user._id,
+      "Library Owner ID"
+    );
+  } catch (e) {
+    errors.push(e);
+  }
+  try {
+    // TODO: THIS WILL BE UPDATED BECAUSE THE WAY OF SERVY CHANGING
+    newLibraryData.fullness = parseInt(newLibraryData.fullness);
+    newLibraryData.fullness = validation.isValidNumber(
+      newLibraryData.fullness,
+      "Fullness Rating"
+    );
+    if (0 > newLibraryData.fullness || newLibraryData.fullness > 5) {
+      throw "Fullness rating must be between 0-5";
     }
-  });
+  } catch (e) {
+    errors.push(e);
+  }
+  try {
+    // TODO:THIS WILL BE UPDATED BECAUSE THE WAY OF SERVY CHANGING
+    newLibraryData.genres = validation.checkStringArray(
+      newLibraryData.genres,
+      "Genres Available"
+    );
+  } catch (e) {
+    errors.push(e);
+  }
+  // TODO: Need to add validation of req.file.path 
+  if (errors.length > 0) {
+    res.render("libraries/new", {
+      errors: errors,
+      hasErrors: true,
+      library: newLibraryData,
+      title: "Creating a Library",
+      id: "NEED TO FIX -> req.user._id",
+    });
+    return;
+  }
+  console.log(req.file)
+  console.log(process.env.DOMAIN+req.file.path)
+  try {
+    const { name, lat, lng, image, fullness } = newLibraryData;
+    const newLibrary = await libraryData.create(
+      name,
+      lat, 
+      lng,
+      process.env.DOMAIN+req.file.path,
```

Comment:
can we add an if condition before this, so if we forget to add env vars we will know before and will save a lot of our debug time